src/components/Task/index.tsx

```diff
@@ -0,0 +1,133 @@
+import React, { ChangeEvent, useState } from "react";
+import { completeIcon, deleteIcon, editIcon, incompleteIcon } from "../Icons";
+import PriorityType from "../../types/PriorityType";
+import TaskType from "../../types/TaskType";
+
+interface TaskProps {
+  task: TaskType;
+  deleteTask: () => void;
+  editTask: (updatedTask: TaskType) => void;
+  completeTask: () => void;
+}
+
+const Task = ({ task, deleteTask, editTask, completeTask }: TaskProps) => {
+  const [isEditing, setIsEditing] = useState(false);
+  const [editedTask, setEditedTask] = useState(task);
+
+  const priorityColor = () => {
+    switch (task.priority) {
+      case PriorityType.high:
+        return "bg-color-red";
+      case PriorityType.medium:
+        return "bg-color-yellow";
+      case PriorityType.low:
+        return "bg-color-green";
+      default:
+        return "";
+    }
+  };
```

Comment:
there is another one on addTask!, Perhaps we can create a utils folder and put this function inside it?